Autopilot/PathManager/Src/pathModeSelector.cpp

```diff
@@ -15,21 +15,69 @@ PathModeSelector* PathModeSelector::getInstance() {
 
 PathModeSelector::PathModeSelector() : current_mode_enum {MODE_TESTING_NONE} {
     current_mode = nullptr; 
+    first_instr = nullptr;
 }
 
 #else
 
 PathModeSelector::PathModeSelector() : current_mode_enum {MODE_TAKEOFF} {
     current_mode = &TakeoffMode::getInstance(); 
+    first_instr = nullptr;
 }
 
 #endif
 
 void PathModeSelector::execute(Telemetry_PIGO_t telemetry_in, SFOutput_t sensor_fusion_in, IMU_Data_t imu_data_in) {
+    
+    //Check whether the telemetry_in has an editflightPath command. If so, we enqueue it into our InstructionQueue
+    if (telemetry_in.waypointModifyFlightPathCommand != NO_FLIGHT_PATH_EDIT) {        
+        PathModeSelector::enqueueInstruction(telemetry_in);
+    }
+
     current_mode->execute(telemetry_in, sensor_fusion_in, imu_data_in);
 }
 
-void PathModeSelector::setAltitdeAirspeedInput(AltitudeAirspeedInput_t alt_airspeed_input) {     
+bool PathModeSelector::instructionQueueIsEmpty() {
+    if (first_instr == nullptr) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+Telemetry_PIGO_t PathModeSelector::dequeueInstruction() {
+    ///Assumes that Queue is non empty.
+        instructionQueueNode* newFirstInstruction = first_instr->nextInstruction; //Store temporary new head
+        Telemetry_PIGO_t instruction = first_instr->instruction;
+        free(first_instr); //Not sure if this the right way to free variables
```

Comment:
in C++ to allocate and free memory we use the `new` and `delete` keywords: https://www.cplusplus.com/doc/tutorial/dynamic/

Reply:
Also, before you delete the memory at first_instr, you have to first set its `first_instr->nextInstruction` to `nullptr` so you don't accidentally delete the next node as well :))

Reply:
go it thanks!